src/main/java/com/iexec/worker/tee/post/compute/web2/UploaderService.java

```diff
@@ -64,17 +66,22 @@ public static String uploadToIpfsWithIexecProxy(String taskId, String baseUrl, S
 
         HttpEntity<ResultModel> request = new HttpEntity<>(resultModel, headers);
 
-        ResponseEntity<String> response = new RestTemplate().postForEntity(baseUrl, request, String.class);
-
-        if (response.getStatusCode().is2xxSuccessful()) {
-            return response.getBody();
+        HttpStatus statusCode = null;
+        try {
+            ResponseEntity<String> response = new RestTemplate().postForEntity(baseUrl, request, String.class);
+            statusCode = response.getStatusCode();
+
+            if (statusCode.is2xxSuccessful()) {
+                return response.getBody();
+            }
+        } catch (HttpClientErrorException e) {
```

Comment:
I think other exceptions can be thrown, such as `HttpServerErrorException` or `UnknownHttpStatusCodeException` or other (?)

Reply:
Right, let's use `RestClientException` which is a common base class for all error thrown by `postForEntity`.